### ISSUE TITLE
Fire event when container.require calls back?

### ISSUE THREAD

**Issue by jergason** [iammerrick/Squire.js#9]
I am trying to use Squire with some mocha tests and PhantomJS. The tests suites all initialize a Squire container, put in some stubs, and then call `container.require` to require the file to test. They then define the Mocha tests inside the callback to `container.require`.

The problem is, we only want to call `mocha.run()` to start the tests once all the tests are defined. We could put special logic in all our tests to fire events after the tests were defined, but that clutters up the tests with something that isn't really their problem.

Could `container` emit an event once the callback was called so we know when the tests are defined and ready to run? Do you have any other ideas around this problem?

**Comment by joeeames** [comment]
Can you put up a simple gist?

**Comment by jergason** [comment]
This is what our tests are doing right now.

``` JavaScript
// main test runner
define(function(require) {
  require('someTest')
  require('someOtherTest')

 mocha.run()
})
```

``` JavaScript
// someTest.js
define(function(require) {
  var Squire = require('Squire')
  var container = new Squire()
  container.mock('someDep', { hurp: 'durp' })
  container.require(['someThingUnderTest'], function(thingToTest) {
    describe('this thing under test', function() {
      it('does correct stuff', function() {
        assert.equal(thingToTest.dep.hurp, 'durp')
      })
    })
  })
})
```

So the test files are requried into the main test file, but the test files themselves define their test suites asynchronously, so when we call `mocha.run()` in the main test files, the test suites have not been defined yet.

**Comment by jergason** [comment]
Addressed in pull request #10.
